nixos/modules/services/misc/omnom.nix

```diff
@@ -0,0 +1,182 @@
+          Configuration options for the /etc/omnom/config.yml file.
+        '';
+        type = lib.types.submodule {
+          freeformType = settingsFormat.type;
+          options = {
+            app = {
+              debug = lib.mkEnableOption "debug mode";
+              disable_signup = lib.mkEnableOption "restricting user creation";
+              results_per_page = lib.mkOption {
+                type = lib.types.int;
+                default = 20;
+                description = "Number of results per page.";
+              };
+            };
+            db = {
+              connection = lib.mkOption {
+                type = lib.types.str;
+                default = "${cfg.dataDir}/db.sqlite3";
+                description = "Database connection URI.";
+                defaultText = lib.literalExpression ''
+                  "''${config.services.omnom.dataDir}/db.sqlite3"
+                '';
+              };
+              type = lib.mkOption {
+                type = lib.types.str;
+                default = "sqlite";
+                description = "Database type.";
+              };
+            };
+            server = {
+              address = lib.mkOption {
+                type = lib.types.str;
+                default = "127.0.0.1:${toString cfg.port}";
+                description = "Server address.";
+                defaultText = lib.literalExpression ''
+                  "127.0.0.1:''${config.services.omnom.port}"
+                '';
+              };
+              secure_cookie = lib.mkOption {
+                type = lib.types.bool;
+                default = true;
+                description = "Limit cookies to secure channel.";
+              };
+            };
+            storage = {
+              root = lib.mkOption {
+                type = lib.types.str;
+                default = "${cfg.dataDir}/static/data";
+                defaultText = lib.literalExpression ''
+                  "''${config.services.omnom.dataDir}/static/data"
+                '';
+                description = "Where the snapshots are saved.";
+              };
+            };
+          };
+        };
+        default = { };
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.etc."omnom/config.yml".source = settingsFormat.generate "omnom-config.yml" cfg.settings;
+
+    services.omnom.settings = {
+      app.debug = lib.mkDefault cfg.debug;
+      storage.type = lib.mkDefault "fs";
+    };
+
+    systemd.services.omnom = {
+      path = [ pkgs.rsync ];
+
+      serviceConfig = {
+        DynamicUser = isDynamicUser;
+        User = cfg.user;
+        Group = cfg.group;
+        ExecStart = ''
+          ${lib.getExe cfg.package} listen \
+            --config /etc/omnom/config.yml \
+            ${lib.optionalString cfg.debug "-d"}
+        '';
+        StateDirectory = lib.mkIf (cfg.dataDir == "/var/lib/omnom") "omnom";
+        StateDirectoryMode = lib.mkIf (cfg.dataDir == "/var/lib/omnom") "0750";
```

Comment:
After looking at the omnom source code for a bit, the whole mode handling this module does seam a bit unnecessary. While the storage root folder is created with [777](https://github.com/asciimoo/omnom/blob/dbf40c9c50b74335286faea7c5070bba11dced83/storage/fs/fs.go#L97) perm, all snapshot/resource files are created with [600](https://github.com/asciimoo/omnom/blob/dbf40c9c50b74335286faea7c5070bba11dced83/storage/fs/fs.go#L78). IMHO we can completely remove all mode related code from the module.